src/test/services/user.test.ts

```diff
@@ -0,0 +1,246 @@
+      expect(isEventLeaderOrStaff(value)).toBe(expected);
+    });
+  });
+
+  describe("isAdminOrEventManager function", () => {
+    const testCases = generateTestCases("isAdminOrEventManager");
+    it.each(testCases)(`returns $expected for $role`, ({ value, expected }) => {
+      expect(isAdminOrEventManager(value)).toBe(expected);
+    });
+  });
+
+  describe("Role Requirements function", () => {
+    USER_ROLES.forEach((role) => {
+      const user = { role, loggedIn: true };
+
+      it(`should correctly check ${role} role`, () => {
+        const requirementFunction = roleRequirements[role];
+        expect(requirementFunction(user)).toBe(true);
+      });
+    });
+
+    it("should handle null user", () => {
+      const requirementFunction = roleRequirements["STUDENT"];
+      expect(requirementFunction(null)).toBe(false);
+    });
+  });
+
+  describe("extractTeacherName function", () => {
+    it("should return null for null input", () => {
+      const result = extractTeacherName(null);
+      expect(result).toBe(null);
+    });
+
+    it("should return null for undefined input", () => {
+      const result = extractTeacherName(undefined);
+      expect(result).toBe(null);
+    });
+
+    it("should return the formatted teacher name (e.g. `J. Doe`) when given both givenName and familyName", () => {
+      const teacher = { givenName: "John", familyName: "Doe" };
+      const result = extractTeacherName(teacher);
+      expect(result).toBe("J. Doe");
+    });
+
+    it("should return only the familyName if givenName is not provided", () => {
+      const teacher = { familyName: "Doe" };
+      const result = extractTeacherName(teacher);
+      expect(result).toBe("Doe");
+    });
+  });
+
+  describe("schoolNameWithPostcode function", () => {
+    const mockSchool: School = {
+      urn: "12345",
+      name: "Example School",
+      postcode: "AB12 3CD",
+      closed: false,
+      dataSource: "MockDataSource",
+    };
+
+    it("should return the school name and postcode when both are provided", () => {
+      const result = schoolNameWithPostcode(mockSchool);
+      expect(result).toBe(mockSchool.name + ", " + mockSchool.postcode);
+    });
+
+    it("should return the school name if no postcode is provided", () => {
+      const result = schoolNameWithPostcode({ ...mockSchool, postcode: "" });
+      expect(result).toBe(mockSchool.name);
+    });
+  });
+
+  describe("isLoggedIn function", () => {
+    it("should return true for a logged-in user", () => {
+      const result = isLoggedIn({ ...mockUser, loggedIn: true });
+      expect(result).toBe(true);
+    });
+
+    it("should return false for a logged-out user", () => {
+      const result = isLoggedIn({ loggedIn: false });
+      expect(result).toBe(false);
+
+      const resultWithUser = isLoggedIn({ ...mockUser, loggedIn: false });
+      expect(resultWithUser).toBe(false);
+    });
+
+    it("should return false for null or undefined user", () => {
```

Comment:
Small one, but this should be two tests, just because it would be more informative in case of failure. For instance, as it is right now, if the test fails in the first assertion, it will never run the second assertion.

The same applies for the test in the `describe` section below.